<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Chip data from QIMR</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PGS|Consistency</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Introduction</a>
</li>
<li>
  <a href="0_PredictorPrep.html">Predictors</a>
</li>
<li>
  <a href="1_GRP_ChipData.html">HSU Data</a>
</li>
<li>
  <a href="2_QIMR_ChipData.html">QIMR Data</a>
</li>
<li>
  <a href="3_WGS.html">WGS data</a>
</li>
<li>
  <a href="4_LowPassSeq.html">lcWGS data</a>
</li>
<li>
  <a href="5_Consistency.html">Consistency</a>
</li>
<li>
  <a href="10_Imputation_Panel_effect.html">Imputation Panel</a>
</li>
<li>
  <a href="6_MissingSNP.html">Missingness</a>
</li>
<li>
  <a href="7_Benchmarking.html">Benchmarking</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Chip data from QIMR</h1>

</div>


<p><br></p>
<p>CEPH samples were used as technical control in QIMR projects too.
Scott helped to extract all the CEPH samples with around 200 samples,
and imputed them with two imputation servers, Michigan and TopMed. The
data is both useful to compare across other studies, and compare two
different imputation panels.</p>
<p><br></p>
<div id="data-qc-and-imputation" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Data QC and
Imputation</h1>
<p>QIMR have many batches of data and they were processed with standard
QC with MAF &gt;=1%, and flip to hg19 ‘+’ strand (which I know, based on
Will Rayner’s BLAST search results).</p>
<p>For each chip family (GSA, Omni, HapMap-based), Scott processed the
data in steps of:</p>
<ol style="list-style-type: decimal">
<li><p>Identify all of the NA/GM control samples which passed QC, and
the sample IDs which they were read as after extracting from
GenomeStudio.</p></li>
<li><p>Make a tiny PLINK binary-format file which has their raw
genotypes under those IDs (still distinct), but restricted to just the
markers which passed QC in their batch. In the Omni case I added 50
ordinary individuals who passed QC in the same batch, otherwise there
were too few for the Imputation Server to accept the job. and added 200
individuals added to each batch of genotyping for a decent sample
size.(Scott just chose the first 200/batch which were covered by the UQ
data transfer agreement)</p></li>
<li><p>Merge the batches from one chip family for the set of markers
used in the original imputation run (which is the lowest common
denominator of QC-passing markers across a number of batches), and flip
to hg19 ‘+’ strand (which I know, based on Will Rayner’s BLAST search
results).</p></li>
<li><p>Upload to Michigan Imputation Server (3 uploads per chip family :
(a) autosomes; (b) chr X PAR regions; (c) chr X non-PAR
region).</p></li>
<li><p>Uploaded to TOPMED imputation server for TOPMed (r3)
reference</p></li>
<li><p>Download and extract the files and transfer to Tian through
RDM.</p></li>
</ol>
</div>
<div id="post-imputation-processing" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Post-imputation
processing</h1>
<div id="convert-to-plink" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> convert to plink</h2>
<pre class="bash"><code>plink --vcf  $vcffile   --const-fid  --keep-allele-order --make-bed --out $plinkfile  

mkdir -p ${folder}_PLINK/${folder}_PLINK_original_bim
cp ${plinkfile}.bim  ${folder}_PLINK/${folder}_PLINK_original_bim/


Rscript update_TOPMED_SNPs.R  ${plinkfile}.bim  SNP_lists/dbsnp_146_hg38_${chr}.txt  /QRISdata/Q6913/Pipeline/ukb20k_7M_4cM/snp.info
cp ${plinkfile}.bim_updated  ${plinkfile}.bim
</code></pre>
</div>
<div id="update-snp-id" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Update SNP ID</h2>
<p>The purpose of this step is to rename the duplicated SNP IDs, and
fill in an ID for the SNPs without an ID.</p>
<p>Duplicate SNP IDs are distinguished by adding dup and a number, while
keeping the SNP that have the same pair of alleles vs. the predictors
intact.</p>
<p>This R script was used for TopMed imputed data</p>
<pre class="r"><code>library(dplyr)
library(tidyr)
library(data.table)

args=commandArgs(trailingOnly = TRUE)

bimfile=args[1]
pred.file=args[2]

# Read the files
predictor &lt;- read.table(pred.file, header = TRUE, stringsAsFactors = FALSE,
                        col.names = c(&quot;chr&quot;, &quot;SNP&quot;, &quot;cm&quot;, &quot;pos&quot;, &quot;A1&quot;, &quot;A2&quot;, &quot;A1Freq&quot;,  &quot;N&quot;, &quot;blk&quot; ))
bim &lt;- read.table(bimfile, header = FALSE, stringsAsFactors = FALSE,
                  col.names = c(&quot;chr&quot;, &quot;SNP&quot;, &quot;cm&quot;, &quot;pos&quot;, &quot;A1&quot;, &quot;A2&quot;))


# add an index column

bim$index = c(1:nrow(bim))

# Create chrbpAB for bim
bim &lt;- bim %&gt;%
  mutate(A = pmin(toupper(A1), toupper(A2)),
         B = pmax(toupper(A1), toupper(A2)),
         chrbpAB = paste0(&quot;chr&quot;, chr, &quot;:&quot;, pos, &quot;:&quot;, A, &quot;:&quot;, B))

# If &#39;SNP&#39; is &quot;.&quot;, replace it with the value in &#39;chrbpAB&#39;
bim &lt;- bim %&gt;%  mutate(SNP = ifelse(SNP == &quot;.&quot;, chrbpAB, SNP))


# Handle duplicates in bim
predictor &lt;- predictor %&gt;%
  mutate(A = pmin(toupper(A1), toupper(A2)),
         B = pmax(toupper(A1), toupper(A2)))

# Match with predictor and handle duplicates
bim &lt;- bim %&gt;%  left_join(predictor, by = c(&quot;SNP&quot;, &quot;A&quot;, &quot;B&quot;), suffix = c(&quot;&quot;, &quot;.pred&quot;)) 
bim &lt;- bim %&gt;%  mutate(exact_match = !is.na(A1.pred) &amp; !is.na(A2.pred)) 

dup.bim = bim[bim$SNP %in% bim[duplicated(bim$SNP) , &quot;SNP&quot;],]
dup.bim &lt;- dup.bim %&gt;% 
  group_by(SNP) %&gt;% 
  arrange(desc(exact_match), .by_group = T) %&gt;%
  ungroup()

dup.bim = data.frame(dup.bim)

# If there are duplicated SNPs for any reason then label them as _2, _3 etc
temp &lt;- rle(dup.bim$SNP)
temp2 &lt;- paste0(dup.bim$SNP, &quot;_&quot;, unlist(lapply(temp$lengths, seq_len)))
temp2 &lt;- gsub(&quot;_1$&quot;, &quot;&quot;, temp2)
dup.bim$SNP &lt;- temp2

## get the key columns
uniq.bim =  bim[!bim$SNP %in% bim[duplicated(bim$SNP) , &quot;SNP&quot;],]

fixed.bim = rbind(uniq.bim %&gt;%  select(chr, SNP, cm, pos, A1, A2, index),
                  dup.bim %&gt;%  select(chr, SNP, cm, pos, A1, A2, index))

fixed.bim = fixed.bim[order(fixed.bim$index),]

# Save the modified bim file
write.table(x = fixed.bim[,1:6], file = paste0(bimfile,&quot;_updated&quot; ), sep = &quot;\t&quot;, quote = FALSE, row.names = FALSE, col.names = FALSE)</code></pre>
<p>HRC imputed data were processed in the same way as HSU data.</p>
</div>
<div id="merge-the-chromosomes" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> merge the
chromosomes</h2>
<p>HRC imputed data were merged into one file from all the
chromosomes.</p>
<pre class="bash"><code>i=$SLURM_ARRAY_TASK_ID
cd /scratch/project/genetic_data_analysis/uqtlin5/CEPH_samples/QIMR
dir=Cleaned_Plink/

## define data
studies=(&quot;GSA&quot; &quot;Hap_HRCr1.1&quot; &quot;Omni_HRCr1.1&quot;)
panel=&quot;HRCr1.1&quot;

for cohort in &quot;${studies[@]}&quot;; do
    cohort=${studies[i-1]}
    ## generate the merge list
    prefix=${dir}/${cohort}_${panel}_PLINK/${cohort}_${panel}&quot;_chr&quot;    
    for chr in $(seq 2 22); do
        echo &quot;${prefix}${chr}.bed ${prefix}${chr}.bim ${prefix}${chr}.fam&quot;  &gt;&gt;  ${cohort}_${panel}_bmerge_files.txt
    done    
    ## merge the data
    mkdir -p Merged_plink
    plink --bfile ${prefix}1  --merge-list ${cohort}_${panel}_bmerge_files.txt  --allow-no-sex  --make-bed  --out  Merged_plink/${cohort}_${panel}_autosomes
done
</code></pre>
<p>TopMed imputed data were filtered with the 7.3M SNP in the SBayesRC
predictors, and then merged onto one file.</p>
<pre class="bash"><code>i=$SLURM_ARRAY_TASK_ID
dir=Cleaned_Plink/

## define data
studies=(&quot;GSA_TOPMedr3&quot; &quot;Hap_TOPMedr3&quot; &quot;Omni_TOPMedr3&quot;)

mkdir -p Cleaned_Plink/${cohort}_PLINK/
bfile=/QRISdata/Q5740/QIMR_Berghofer/${cohort}_PLINK/${cohort}_chr${i}
plink --bfile $bfile --extract  SNP_lists/SNPs_in_7.3M.txt   --make-bed --out Cleaned_Plink/${cohort}_PLINK/${cohort}_subset_chr${i}

for cohort in &quot;${studies[@]}&quot;; do
cohort=${studies[i-1]}
## generate the merge list
prefix=${dir}/${cohort}_PLINK/${cohort}&quot;_subset_chr&quot;    
for chr in $(seq 2 22); do
    echo &quot;${prefix}${chr}.bed ${prefix}${chr}.bim ${prefix}${chr}.fam&quot;  &gt;&gt;  ${cohort}_bmerge_files.txt
done    
## merge the data
mkdir -p Merged_plink
plink --bfile ${prefix}1  --merge-list ${cohort}_bmerge_files.txt  --allow-no-sex  --make-bed  --out  Merged_plink/${cohort}_autosomes
done
</code></pre>
</div>
</div>
<div id="pgs-profiling" class="section level1" number="3">
<h1><span class="header-section-number">3</span> PGS profiling</h1>
<p>The same plink command line was used to profile PGS from the 6 QIMR
data sets.</p>
<pre class="bash"><code>plink  \
--bfile  $bfile \
--score  ${predictor}  2 5 8  header sum    \
--out  $output</code></pre>
<p>The parameters after your predictor file means</p>
<ul>
<li>2 5 8: the three useful columns in the predictor file.<br />
</li>
<li>header: The predictor file has a header row.<br />
</li>
<li>sum: Plink prefers to divide the score by the number of SNPs in
predictor. Using “sum” will prevent the division step.</li>
</ul>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
